crates/katana/node-bindings/src/lib.rs

```diff
@@ -428,56 +441,92 @@ impl Katana {
         if let Some(db_dir) = self.db_dir {
             cmd.arg("--db-dir").arg(db_dir);
         }
-        if let Some(rpc_url) = self.rpc_url {
-            cmd.arg("--rpc-url").arg(rpc_url);
+
+        if let Some(url) = self.l1_provider {
+            cmd.arg("--l1.provider").arg(url);
         }
 
+        // Need to make sure that the `--dev` is not being set twice.
```

Comment:
Maybe a logic with `HashSet` could make the flow simpler and at the end every unique value is added to the `cmd`.

Reply:
`HashSet` doesn't maintain insertion order. there's [`IndexSet`](https://crates.io/crates/indexset) but the flow itself is already pretty simple.